### ISSUE TITLE
Epoch vs Steps_per_epoch vs Batch_size

### ISSUE THREAD

**Issue by Densus100** [matterport/Mask_RCNN#2848]
Hello guys, I'm trying to re-train the Mask RCNN to fit into my dataset, but i stuck at the config and really confused about some parameters. **Please anyone that maybe able to help me distinguish the difference between :**

1. EPOCH
2. STEPS_PER_EPOCH
3. BATCH_SIZE                      

Explanation with example like the numbers of the dataset and how many iteration that the models do between those parameters will be much appreciated. Thank you :)

**Comment by raulperezalejo** [comment]
Hello, this is my understanding: 

EPOCH and STEPS_PER_EPOCH:
 can be found [here](https://keras.io/api/models/model_training_apis/#fit-method) as an input parameter of the `fit` method.

BATCH_SIZE:
This is calculates this way: BATCH_SIZE =IMAGES_PER_GPU * GPU_COUNT
GPU_COUNT is simply the amount of GPU you have, for example is colab is only 1
IMAGES_PER_GPU: is the amount of images the CPU is going to process each time. In my case ill send my model one picture each time, in this case i set this number to 1

hope it helps

**Comment by Densus100** [comment]
Thank you @raulperezalejo
